aerogear/aerogear-sync-server
# Server/Client Engine should support JSON Patch

Changes required to support JSON Patch.

How to test:
1. [Start](https://github.com/danbev/aerogear-sync-server/tree/refactoring-squashed/server/server-netty#jsonpatchsyncserver) the JsonPatchSyncServer. 
2. Check out @lholmquist [aerogear-js-cookbook](https://github.com/lholmquist/aerogear-js-cookbook/tree/diff-sync-json-patch) branch. Follow the instructions to start the ember server.
3. Check out my Android demo branch [json-patch-sync-server-example](https://github.com/danbev/android-diffsync-demo/tree/json-patch-sync-server-example) and follow the instructions in the readme to build and deploy.

It is not necessary to do both steps _2_ an _3_, either one would be fine. 

JIRAs:
[AGSYNC-24 "Server Engine should support JSON Patch"](https://issues.jboss.org/browse/AGSYNC-24) 
[AGSYNC-25 "Java Client Engine should support JSON Patch"](https://issues.jboss.org/browse/AGSYNC-25) 
These are part of the epic [AGSYNC-9 "Allow the Sync Server exchange JSON patches"](https://issues.jboss.org/browse/AGSYNC-9)

@lholmquist @secondsun Would you give mind to review? (sorry about the size of this PR)
